source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1645,19 +1645,24 @@ void Filter::closeGrpcStreamIfLastRespReceived(const ProcessingResponse& respons
     return;
   }
 
+  if (decoding_state_.callbackState() != ProcessorState::CallbackState::Idle ||
```

Comment:
checking both states in one method seems confusing.
OTOH, does this work for observability mode?